app/forms/supplementary/DeclarationAdditionalActors.scala

```diff
@@ -16,23 +16,23 @@
 
 package forms.supplementary
 
-import forms.MetadataPropertiesConvertable
 import play.api.data.Forms.{optional, text}
 import play.api.data.{Form, Forms}
-import play.api.libs.json.Json
+import play.api.libs.json.{JsValue, Json}
 import uk.gov.voa.play.form.ConditionalMappings.mandatoryIfNot
 
-case class DeclarationAdditionalActors(eori: Option[String], partyType: Option[String])
-    extends MetadataPropertiesConvertable {
+case class DeclarationAdditionalActors(eori: Option[String], partyType: Option[String]) {
+  implicit val writes = Json.writes[DeclarationAdditionalActors]
 
-  override def toMetadataProperties(): Map[String, String] =
-    Map(
-      "declaration.goodsShipment.aeoMutualRecognitionParties[0].id" -> eori.getOrElse(""),
-      "declaration.goodsShipment.aeoMutualRecognitionParties[0].roleCode" -> partyType.getOrElse("")
-    )
+  def isDefined: Boolean = eori.isDefined && partyType.isDefined
+
+  def toJson: JsValue = Json.toJson(this)
 }
 
 object DeclarationAdditionalActors {
+
+  def fromJson(value: JsValue): DeclarationAdditionalActors = Json.fromJson(value).get
```

Comment:
I don't like `get`. Probably it's `Option` so to avoid problems with empty data my propose is to use `getOrElse("")`